### ISSUE TITLE
Can not correctly read abnormal terminated OUTCAR

### ISSUE THREAD

**Issue by tw2469** [Ionizing/rsgrad#12]
thread 'main' panicked at src/vasp_parsers/outcar.rs:181:9:
assertion `left == right` failed: Init failed due to incomplete OUTCAR: cputimev.len() = 37 != 38
  left: 37
 right: 38
stack backtrace:
   0:     0x55c0f6022215 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h1e1a1972118942ad
   1:     0x55c0f604b44b - core::fmt::write::hc090a2ffd6b28c4a
   2:     0x55c0f601f35f - std::io::Write::write_fmt::h8898bac6ff039a23
   3:     0x55c0f6021fee - std::sys_common::backtrace::print::ha96650907276675e
   4:     0x55c0f6023459 - std::panicking::default_hook::{{closure}}::h215c2a0a8346e0e0
   5:     0x55c0f602319d - std::panicking::default_hook::h207342be97478370
   6:     0x55c0f60238f3 - std::panicking::rust_panic_with_hook::hac8bdceee1e4fe2c
   7:     0x55c0f60237d4 - std::panicking::begin_panic_handler::{{closure}}::h00d785e82757ce3c
   8:     0x55c0f60226d9 - std::sys_common::backtrace::__rust_end_short_backtrace::h1628d957bcd06996
   9:     0x55c0f6023507 - rust_begin_unwind
  10:     0x55c0f5c2b2c3 - core::panicking::panic_fmt::hdc63834ffaaefae5
  11:     0x55c0f5c2b6df - core::panicking::assert_failed_inner::hda4754f94c1c1cb1
  12:     0x55c0f5c2069f - core::panicking::assert_failed::h16f920584301709e
  13:     0x55c0f5d66a31 - rsgrad::vasp_parsers::outcar::Outcar::from_file::h32aed19639b7a02c
  14:     0x55c0f5d5b198 - <rsgrad::commands::rlx::Rlx as rsgrad::cli::OptProcess>::process::h618ef540f27ab766
  15:     0x55c0f5c2dc3f - rsgrad::cli::run::h37828720236563e5
  16:     0x55c0f5c2d78b - rsgrad::main::h55a6b52e627803a9
  17:     0x55c0f5c2bc63 - std::sys_common::backtrace::__rust_begin_short_backtrace::h60afcb835f1104fb
  18:     0x55c0f5c2bafd - std::rt::lang_start::{{closure}}::h8a1ca9bd1beea3f1
  19:     0x55c0f60167e0 - std::rt::lang_start_internal::h3ed4fe7b2f419135
  20:     0x55c0f5c2d8a5 - main
  21:     0x7f63dd295555 - __libc_start_main
  22:     0x55c0f5c2b9f0 - <unknown>
  23:                0x0 - <unknown>

[OUTCAR.txt](https://github.com/user-attachments/files/16431716/OUTCAR.txt)

**Comment by Ionizing** [comment]
It seems `rsgrad rlx` fails at those checkings:

https://github.com/Ionizing/rsgrad/blob/9c9d697c8f13c38abfa2b932ddc0ae2231b090af/src/vasp_parsers/outcar.rs#L177-L184

```bash
> rg "free  energy" OUTCAR.txt | wc -l
      38
> rg "LOOP\+" OUTCAR.txt | wc -l
      37
```

The `LOOP+ ...` is at the end of each ionic loop and the last one is missing:

```
--------------------------------------------------------------------------------------------------------


    POTLOK:  cpu time    0.1757: real time    0.1757


--------------------------------------------------------------------------------------------------------




--------------------------------------------------------------------------------------------------------


    WAVPRE:  cpu time    0.0564: real time    0.0570
    FEWALD:  cpu time    0.0016: real time    0.0016
    ORTHCH:  cpu time   18.4224: real time   18.4353
     LOOP+:  cpu time 1253.5412: real time 1255.4302


--------------------------------------- Iteration     38(   1)  ---------------------------------------


    POTLOK:  cpu time    0.1499: real time    0.1502
    SETDIJ:  cpu time    0.0239: real time    0.0239
atom =  81  type =  3  l = 2

......
TOO MUCH TEXT OMITTED
......

--------------------------------------------------------------------------------------------------------


    POTLOK:  cpu time    0.1783: real time    0.1784


--------------------------------------------------------------------------------------------------------


[END OF FILE]
```

Since the `cputime` output comes at the end of each ionic loop and it doesn't affect the data already parsed, it's acceptable to just pad it with zeros and log a warn to user.

**Comment by Ionizing** [comment]
Fixed via d0bda96a25de62d37b6839e882eb2bd6231acfb1 .
